### PR TITLE
chore: Rename data packages -> datasets

### PR DESCRIPTION
**Pull request by spencerwilson**
- Throughout Rust code, rename data packages to datasets where appropriate (that is, where datasets are what's being referred to rather than data packages)
- Rename datapackage.json -> dataset.json
- Also rename the struct `DataResource` -> `Table`
- Update all the docstrings
- `init`: Change `-p`/`--package <NAME>` argument to `-n`/`--name <NAME>`

All my ❤️ to the rust-analyzer devs whose work made this task just doing "right-click > Rename Symbol" a hundred times.

Note that these renames do _not_ extend into the contents of data packages; generated Node.js/Python/C# code is unchanged. This means that one examines a data package they may see outdated terminology in use. We can correct later if valuable.

### CODE REVIEW

**Comment by spencerwilson** [comment]
Current dependencies on/for this PR:
* main
  * **PR #198** <a href="https://app.graphite.dev/github/pr/patch-tech/dpm/198?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #199** <a href="https://app.graphite.dev/github/pr/patch-tech/dpm/199?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #200** <a href="https://app.graphite.dev/github/pr/patch-tech/dpm/200?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #201** <a href="https://app.graphite.dev/github/pr/patch-tech/dpm/201?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #202** <a href="https://app.graphite.dev/github/pr/patch-tech/dpm/202?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #203** <a href="https://app.graphite.dev/github/pr/patch-tech/dpm/203?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/patch-tech/dpm/203?utm_source=stack-comment).

File: src/codegen/generator.rs [hunk 1]
```diff
@@ -55,14 +55,14 @@ pub fn exec_cmd(name: &str, path: &Path, cmd: &str, args: &[&str]) {
     }
 }
 
-/// A type that derives the contents of a data package from a `DataPackage` descriptor.
+/// A type that derives the contents of a data package from a `Dataset`.
 pub trait Generator {
     /// The data package that the generator is processing.
```
Comment (ajmasci) [inline src/codegen/generator.rs:60]: Doesn't the docstring also need to change?
```suggestion
    /// The dataset that the generator is processing.
```

Comment (spencerwilson) [reply src/codegen/generator.rs:60]: yes!

File: src/codegen/generator.rs [hunk 2]
```diff
@@ -87,7 +87,7 @@ pub trait Generator {
     fn file_name(&self, name: &str) -> String;
 
     /// Returns a package name in the language, given a name.
```
Comment (ajmasci) [inline src/codegen/generator.rs:89]: Same here?

Comment (spencerwilson) [reply src/codegen/generator.rs:89]: Nope, this refers to the name of the generated data package

**Review by whelanBoyd — APPROVED** [review]

File: src/codegen/generator.rs (hunk 1 above)
Comment (spencerwilson) [reply src/codegen/generator.rs:60]: Fixed in next-upstack PR.

**Pull request merged**